src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,91 @@
+package com.ably.kafka.connect.client;
+
+import com.ably.kafka.connect.config.ChannelSinkConnectorConfig;
+import com.ably.kafka.connect.config.ConfigValueEvaluator;
+import com.ably.kafka.connect.mapping.ChannelSinkMapping;
+import com.ably.kafka.connect.mapping.MessageSinkMapping;
+import io.ably.lib.http.HttpCore;
+import io.ably.lib.http.HttpUtils;
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.HttpPaginatedResponse;
+import io.ably.lib.types.Message;
+import io.ably.lib.types.Param;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * Ably Batch client based on REST API
+ */
+public class DefaultAblyBatchClient extends DefaultAblyClient {
```

Comment:
`DefaultAblyClient` still has a realtime connection open, so it'd be better if we could just remove it entirely if the goal here is to switch to REST publishing. We don't really want the complexity of there being an idle WebSocket connection hanging around

Reply:
Thanks removing references to `DefaultAblyClient`, so that the `DefaultAblyBatchClient` can just implement the `AblyClient` interface